nrfconnect/sdk-nrf
# add support for nrf53 network core firmware upgrade

Add support for nrf53 network core updates.
TLDR the update is validated by the app core mcuboot, and then copied over to the network core by the network core bootloader through a the `pcd` subsystem.

Functionality for updating depends on https://github.com/nrfconnect/sdk-mcuboot/pull/118 but the PR stands by itself.

Note that the RAM section being locked by mcuboot will be made unavailable in the linker script of the app through this pr:
https://github.com/nrfconnect/sdk-nrf/pull/2080

### To test:

```
west build -b nrf5340pdk_nrf5340_cpuapp -d build_netboot samples/bluetooth/peripheral_uart -f -- -DCONFIG_BOOTLOADER_MCUBOOT=y
west flash -d build_netboot
nrfjprog --program build_netboot/zephyr/net_core_app_moved_test_update.hex --sectorerase -r
```

Verify that the UART output mentions:
```
[00:00:00.020,660] <inf> mcuboot: Boot source: none
[00:00:00.026,092] <inf> mcuboot: Swap type: test
[00:00:00.276,641] <inf> mcuboot: Starting network core update
[00:00:09.978,851] <inf> mcuboot: Done updating network core

```

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
